src/src/PluginStructs/P104_data_struct.cpp

```diff
@@ -428,41 +464,54 @@ void P104_data_struct::createHString(char *pH,
 # endif // ifdef P104_USE_NUMERIC_DOUBLEHEIGHT_FONT
 
 bool P104_data_struct::handlePluginOncePerSecond(taskIndex_t taskIndex) {
-  bool retval = false;
+  bool retval     = false;
+  bool newFlasher = !flasher;
+
+  for (auto it = zones.begin(); it != zones.end(); ++it) {
+    retval = false;
 
-  for (auto it = zones.begin(); it != zones.end() && !retval; ++it) {
     switch (it->content) {
       case 1: // time
+      case 2: // time sec
       {
-        getTime(szTimeL, flasher);
-
-        # ifdef P104_USE_NUMERIC_DOUBLEHEIGHT_FONT
-
-        if (it->layout == 2) {
-          createHString(szTimeH, szTimeL);
-          displayOneZoneText(it->zone - 1, *it, String(szTimeH));
-        } else
-        # endif // ifdef P104_USE_NUMERIC_DOUBLEHEIGHT_FONT
-        {
-          displayOneZoneText(it->zone - 1, *it, String(szTimeL));
-        }
-        flasher = !flasher;
+        getTime(szTimeL, it->content == 2, flasher);
+        flasher = newFlasher;
         retval  = true;
         break;
       }
-      case 2:  // date/4
+      case 3: // date/4
+      case 4: // date/6
+      case 5: // date/7
       {
+        if (lastDay != node_time.day()) {
+          getDate(szTimeL, it->content != 3, it->content == 5);
+          retval  = true;
+          lastDay = node_time.day();
+        }
         break; // TODO
       }
-      case 3:  // date/6
+      case 6:  // date-time/9
```

Comment:
Why not make defines for it, or an enum, if you need to keep the numbers in sync and need to comment it every time you use it?

Reply:
This has quite some room for improvement, indeed. Will fix.